### DISCUSSION TITLE
Click analytics

### DISCUSSION THREAD

**Discussion by garethhayter** [Lumieducation/Lumi#1575] [Q&A]
Hi. I want to create interactive videos with branching (similar to choose your own adventure books) where users can choose which content to view next. In my case, it will be product videos where users can chose which feature to view next. I want to log which choices they clicked and in what order. I want to incorporate this with Google Analytics for example, to see what choices my users prefer. Is this possible?

**Comment by sr258** [comment]
H5P emits xAPI statements that could be converted to other events like calls to Google Analytics. This requires dome programming knowledge. If you have that, we can give you some hints what to do.

**Comment by garethhayter** [comment]
Thanks Sebastian. Yes, I am a developer. Please can you send me the other details. Many thanks 😉

**Comment by sr258** [comment]
Sorry for the delay. I was on holiday...

H5P emits the xAPI statements in a global event handler, that you can subscribe to by adding this JS to the page: 

```
        H5P.externalDispatcher.on('xAPI', (event) => {
            // add your logic to send the event to the LRS  here. 
            console.log('xAPI event: ', event); //(I am just logging to browser developer console for demo purpose)
        });
```

This script should be executed at some point after the H5P core was initialized. It's a good idea to add it as the last script to the head.

**Comment by garethhayter** [comment]
Thanks Sebastian. This looks great. I’ll let you know how I go.
